### PR TITLE
Convert project to ts if not already

### PR DESCRIPTION
**Pull request by whostolebenfrog**

### CODE REVIEW

File: .atomist/editors/AddTypeScript.ts [hunk 1]
```diff
@@ -25,6 +25,11 @@ class AddTypeScript implements EditProject {
 
     edit(project: Project) {
         if (!project.fileExists(".atomist/manifest.yml")) {
+            console.log("Can't run as this doesn't appear to be a rug archive")
```
Comment (ddgenome) [inline .atomist/editors/AddTypeScript.ts]: I like this.

File: .atomist/editors/AddTypeScript.ts [hunk 2]
```diff
@@ -25,6 +25,11 @@ class AddTypeScript implements EditProject {
...
+            return;
+        }
+
+        if (project.fileExists(".atomist/package.json")) {
```
Comment (ddgenome) [inline .atomist/editors/AddTypeScript.ts]: Should we send a console message here too? "This is project is already TypeScript ready"

File: .atomist/editors/AddTypeScriptEditor.ts [hunk 3]
```diff
@@ -49,10 +49,14 @@ class AddTypeScriptEditor implements EditProject {
...
 
     edit(project: Project) {
-        if (!IsRugArchive(project) || !IsSetUpForTypeScript(project)) {
+
+        if (!IsRugArchive(project)) {
+            console.log("Can't run when not a rug archive");
             return;
         }
 
+        project.editWith("AddTypeScript", {});
```
Comment (ddgenome) [inline .atomist/editors/AddTypeScriptEditor.ts]: I'm not sure this is a good idea. If we decide it is, we should made the change to all the AddTypeScript* editors.

File: .atomist/editors/AddTypeScript.ts (hunk 2 above)
Comment (jessitron) [inline .atomist/editors/AddTypeScript.ts]: We thought about that but the point is to make it idempotent.

File: .atomist/editors/AddTypeScriptEditor.ts (hunk 3 above)
Comment (jessitron) [inline .atomist/editors/AddTypeScriptEditor.ts]: I think this is a good idea, very strongly.
Why would you not?

Why fail when we can implement the prerequisites for them?

**Comment by whostolebenfrog** [comment]
@ddgenome Well our thinking was that if you were trying to add a TypeScript editor to a project you would always need the project to be a valid TypeScript archive first. So we updated it make it one first then added a quick change to make AddTypeScript idempotent.

Would you favour that being two steps?

**Comment by ddgenome** [comment]
I could be convinced either way. If we go the route of adding the TypeScript files automatically, then we should do it for all the AddTypeScript* editors.

**Comment by ddgenome** [comment]
I've updated to make everything consistent and fixed a couple things I noticed in existing code.

**Comment by whostolebenfrog** [comment]
Awesome - thanks @ddgenome !

**Pull request merged**